packages/runtime/src/runtime.ts

```diff
@@ -336,32 +347,38 @@ export class Runtime {
         : asaDef.metadataHash,
       mockSuggestedParams(flags, this.round)
     );
+
+    if (this.ctx.tx === undefined || this.ctx.tx.type !== "acfg") { // could already be defined (if used as a txGroup in this.executeTx())
```

Comment:
what's "acfg"?

Reply:
It's a type of transaction (asset deploy/modify/delete ..): [https://developer.algorand.org/docs/get-details/dapps/avm/teal/specification/#typeenum-constants](https://developer.algorand.org/docs/get-details/dapps/avm/teal/specification/#typeenum-constants)

I think using an Enum here would be more intuitive. Updating